# Unit Economics with PythonAssignment
Calculate Unit Economics for "TechStream Solutions" including:

CAC

ARPU

COGS

Gross Margin

LTV

LTV / CAC

## 1. CAC
### 1.1. Monthly Expense



In [ ]:
import pandas as pd

url = 'https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/export?format=xlsx'
monthly_expense = pd.read_excel(url, sheet_name='Sheet1')

In [ ]:
# Check data
monthly_expense.sample(10)

#      month           category             item  amount
0    1 2023-01-01       Server Costs      AWS Hosting    8000
13  14 2023-02-01  Software Licenses       Salesforce    1600
4    5 2023-01-01  Software Licenses       Salesforce    1500
9   10 2023-02-01       Server Costs      AWS Hosting    8200
14  15 2023-02-01  Software Licenses             Zoom     520
18  19 2023-03-01       Server Costs      AWS Hosting    8400
25  26 2023-03-01              Other  Office Supplies     600
2    3 2023-01-01  Software Licenses   Atlassian Jira    1200
8    9 2023-01-01              Other  Travel Expenses    3000
5    6 2023-01-01  Software Licenses             Zoom     500

### Filtering the last month
to calculate the unit for the same month (simple way)

In [ ]:
last_month = monthly_expense['month'].max().month
last_month

3

In [ ]:
cond = monthly_expense['month'].dt.month == last_month
last_month_expense = monthly_expense[cond]
last_month_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

### Retrieve only sales and marketing expense

To calculate the CAC, only sales and marketing expense are taken into account which inclue License for Salesforce (CRM system)

In [ ]:
crm_cost = last_month_expense[last_month_expense['item']=='Salesforce']['amount'].values[0]
crm_cost

1700

Let's add this `crm_cost` as the initial value of `TotalExpense`

In [ ]:
TotalExpense = crm_cost

### 1.2. Salaries

In the customer acquisition cost, we will include salaries of the employees that involved in.

The data **Payroll** has that salary information.

https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/edit?gid=0#gid=0


In [ ]:
url_payroll = 'https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/export?format=xlsx'
payroll = pd.read_excel(url_payroll, sheet_name='Sheet1')

In [ ]:
payroll.sample(10)

month   department  employee_name            position  paid
4  2023-01-01    Marketing  Alice Johnson   Marketing Manager  1650
26 2023-02-01           HR    Frank Green          HR Manager  1300
10 2023-01-01           HR     Gary Black       HR Specialist   700
24 2023-02-01  Engineering      Diana Lee           Developer  1700
11 2023-01-01      Support   Hannah Scott        Support Lead   600
31 2023-02-01    Analytics  Michael Brown   Analytics Manager  1700
30 2023-02-01      Support    Emily Clark  Support Specialist   450
28 2023-02-01      Support   Hannah Scott        Support Lead   600
32 2023-02-01    Analytics   Sarah Wilson        Data Analyst  1200
48 2023-03-01    Analytics  Michael Brown   Analytics Manager  1700

### Last month salaries

In [ ]:
last_month = payroll['month'].max().month
last_month

3

In [ ]:
last_month_payroll = payroll[payroll['month'].dt.month == last_month]
last_month_payroll

month   department   employee_name            position  paid
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

### Sales and Marketing specialists

In [ ]:
sales_mkt_pos = ['Sales Associate','Content Specialist']
last_month_payroll_sales_mkt = last_month_payroll[last_month_payroll['position'].isin(sales_mkt_pos)]
last_month_payroll_sales_mkt

month department employee_name            position  paid
35 2023-03-01      Sales    Jane Smith     Sales Associate   600
36 2023-03-01      Sales     Jim Brown     Sales Associate   700
37 2023-03-01      Sales  Laura Miller     Sales Associate   800
39 2023-03-01  Marketing     Bob Davis  Content Specialist   700

### Sales and Marketing expense including Management

The inclusion of sales and marketing teams' managers in the CAC depends on their role and how their direct activities contribute to acquiring new customers. Let's assume our managers are directly involved into the process.

Therefore, we need to calculate their salaries for CAC.

In [ ]:
sales_mkt_pos = ['Sales Associate','Content Specialist', 'Sales Manager', 'Marketing Manager']
last_month_payroll_sales_mkt = last_month_payroll[last_month_payroll['position'].isin(sales_mkt_pos)]
last_month_payroll_sales_mkt

month department  employee_name            position  paid
34 2023-03-01      Sales       John Doe       Sales Manager  1500
35 2023-03-01      Sales     Jane Smith     Sales Associate   600
36 2023-03-01      Sales      Jim Brown     Sales Associate   700
37 2023-03-01      Sales   Laura Miller     Sales Associate   800
38 2023-03-01  Marketing  Alice Johnson   Marketing Manager  1650
39 2023-03-01  Marketing      Bob Davis  Content Specialist   700

In [ ]:
# Extract the paid values
last_month_payroll_sales_mkt['paid'].sum()

5950

... and add to `TotalExpense`

In [ ]:
TotalExpense += last_month_payroll_sales_mkt['paid'].sum()
TotalExpense

7650

## 1.3. Marketing Spending

In the customer acquisition cost, we will include marketing spending.

The data **daily_marketing_spendings** has that salary information.

https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/edit?gid=1513891606#gid=1513891606

In [ ]:
url_mkt = 'https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/export?format=xlsx'
mkt_expense = pd.read_excel(url_mkt, sheet_name='Sheet1')

In [ ]:
mkt_expense.sample(10)

date       channel  spending
8   2023-01-03    Google Ads       109
26  2023-01-07  LinkedIn Ads       777
239 2023-03-01   Twitter Ads       986
118 2023-01-30  LinkedIn Ads       790
108 2023-01-28    Google Ads       405
171 2023-02-12   Twitter Ads       386
284 2023-03-13    Google Ads       871
253 2023-03-05  Facebook Ads       655
125 2023-02-01  Facebook Ads       468
312 2023-03-20    Google Ads       289

### Last month marketing expense


In [ ]:
last_month = mkt_expense['date'].max().month

last_month_mkt_expense = mkt_expense[mkt_expense['date'].dt.month==last_month]
last_month_mkt_expense


date       channel  spending
236 2023-03-01    Google Ads       449
237 2023-03-01  Facebook Ads       229
238 2023-03-01  LinkedIn Ads       835
239 2023-03-01   Twitter Ads       986
240 2023-03-02    Google Ads       912
..         ...           ...       ...
355 2023-03-30   Twitter Ads       960
356 2023-03-31    Google Ads       143
357 2023-03-31  Facebook Ads       183
358 2023-03-31  LinkedIn Ads       533
359 2023-03-31   Twitter Ads       909

[124 rows x 3 columns]

Let's add `TotalExpense`

In [ ]:
TotalExpense += last_month_mkt_expense['spending'].sum()
TotalExpense

76480

## 1.4. Number of customers

To get the CAC, we need to calculate the new customers

The data **receipts_history** has that salary information.

https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/edit?gid=952917703#gid=952917703

In [ ]:
url_receipt = 'https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/export?format=xlsx'
receipt = pd.read_excel(url_receipt, sheet_name='Sheet1')

In [ ]:
receipt.sample(10)

date  customer_id  receipt_amount  new_customer
418 2023-02-10         1698             304             1
526 2023-02-20         1082             142             0
796 2023-03-19         1943             346             0
235 2023-01-24         2812             230             0
226 2023-01-23         2556             395             1
313 2023-02-03         2940              41             0
278 2023-01-29         2870             491             1
813 2023-03-20         1413             287             0
199 2023-01-20         1766             412             0
395 2023-02-08         2970             123             0

### Last month of receipt

In [ ]:
last_month = receipt['date'].max().month

last_month_receipt = receipt[receipt['date'].dt.month==last_month]
last_month_receipt

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

### Extracting only new customers

In [ ]:
last_month_receipt_new_customer = last_month_receipt[last_month_receipt['new_customer']==1]

In [ ]:
last_month_receipt_new_customer.sample(10)

date  customer_id  receipt_amount  new_customer
895 2023-03-27         2791             222             1
712 2023-03-08         2291             214             1
806 2023-03-20         2588             348             1
855 2023-03-23         1862             226             1
845 2023-03-22         2462             155             1
723 2023-03-10         2173             127             1
692 2023-03-07         1544             335             1
777 2023-03-17         1360              89             1
805 2023-03-20         1667             434             1
725 2023-03-11         1703              46             1

### Number of New Customers

In [ ]:
NumberOfCustomers = len(last_month_receipt_new_customer)
NumberOfCustomers

63

## 1.5. Calculate CAC

In [ ]:
CAC = TotalExpense / NumberOfCustomers
CAC

1213.968253968254

So the CAC is about **$1213.97**

## 2. Average Revenue Per User (ARPU)

Total Revenue is extracted from receipt information

In [ ]:
TotalRevenue = last_month_receipt_new_customer['receipt_amount'].sum()

In [ ]:
ARPU = TotalRevenue / NumberOfCustomers
ARPU

274.92063492063494

## 3. Cost of Good Sold (COGS)

From monthly expense, it includes:
- Server
- Software license
- Salary for employees directly involved in the product development

## 3.1. Server and Software license

In [ ]:
last_month_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
server_software_items = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira']
prod_expense = last_month_expense[last_month_expense['item'].isin(server_software_items)]
prod_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400

### Shared expense for software of Slack and Zoom

In [ ]:
shared_items = ['Slack', 'Zoom']
shared_expense = last_month_expense[last_month_expense['item'].isin(shared_items)]
shared_expense

#      month           category   item  amount
21  22 2023-03-01  Software Licenses  Slack     900
23  24 2023-03-01  Software Licenses   Zoom     540

### Final Expense for Server and Software

In [ ]:
server_sofware_expense = prod_expense['amount'].sum() + shared_expense['amount'].sum()
server_sofware_expense

15640

## 3.2. Production team salary

In [ ]:
last_month_payroll

month   department   employee_name            position  paid
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

We will only Engineering department in the process of production

In [ ]:
production_salaries = last_month_payroll[last_month_payroll['department'] == 'Engineering']['paid'].sum()
production_salaries

5200

## 3.3. COGS calculation

In [ ]:
COGS = server_sofware_expense + production_salaries
COGS

20840

## 4. Gross Margin

In [ ]:
Revenue = last_month_receipt['receipt_amount'].sum()

Gross_Margin = (Revenue - COGS)/Revenue * 100
Gross_Margin

74.9015451687883

## 5. Lifetime value (LTV)

In [ ]:
url_lifespan = 'https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/export?format=xlsx'
lifespan = pd.read_excel(url_lifespan, sheet_name='Sheet1')

In [ ]:
lifespan.sample(10)

Unnamed: 0 start_date churn_date
35        1035 2020-06-18 2023-02-28
14        1014 2021-06-27 2021-09-04
32        1032 2022-08-12 2022-09-18
76        1076 2020-01-25 2021-06-19
28        1028 2022-02-13 2022-03-27
11        1011 2022-01-25 2022-09-10
3         1003 2021-08-22 2023-02-07
95        1095 2023-01-09 2023-02-28
59        1059 2020-11-19 2023-03-08
42        1042 2020-01-24 2021-08-16

In [ ]:
# calculate customer lifespan in days
lifespan['ls_days'] = (lifespan['churn_date'] - lifespan['start_date']).dt.days
lifespan.sample(10)

Unnamed: 0 start_date churn_date  ls_days
31        1031 2023-03-13 2023-03-22        9
75        1075 2022-03-23 2023-03-27      369
10        1010 2021-03-24 2023-03-30      736
3         1003 2021-08-22 2023-02-07      534
94        1094 2020-10-17 2021-05-09      204
85        1085 2021-07-01 2021-07-22       21
40        1040 2022-07-06 2023-01-24      202
5         1005 2021-12-06 2022-12-29      388
83        1083 2021-10-25 2022-12-14      415
0         1000 2021-11-15 2022-09-14      303

In [ ]:
# calculate average lifespan in days
avg_lifespan = lifespan['ls_days'].mean()
avg_lifespan

295.24

In [ ]:
# calculate average lifespan in months
avg_lifespan_months = avg_lifespan/30
avg_lifespan_months

9.841333333333333

In [ ]:
LTV = ARPU * avg_lifespan_months * Gross_margin
LTV

202652.54266051034

## 6. LTV / CAC

In [ ]:
LTV_CAC = LTV / CAC
LTV_CAC

166.93397211835972

## **Conclusion and Recommendations**

**Conclusion:**

Based on the calculated Unit Economics for "TechStream Solutions," the key findings are as follows:

- Customer Acquisition Cost (CAC): The CAC is approximately $1213.97, indicating a significant investment in acquiring each customer. This includes expenses for marketing, sales salaries, and associated tools like Salesforce.

- Average Revenue Per User (ARPU): The ARPU reflects the average revenue generated per new customer, showing the profitability potential per user.

- Cost of Goods Sold (COGS): This includes server costs, software licenses, and engineering team salaries. These costs contribute to product development and operational expenses.

- Gross Margin: The calculated Gross Margin provides insights into the company's profitability after deducting COGS from total revenue.

- Lifetime Value (LTV): LTV considers customer lifespan and Gross Margin, highlighting the long-term value a customer brings to the business.

- LTV/CAC Ratio: A high LTV/CAC ratio signifies effective marketing strategies and a profitable customer base.


**Recommendations:**

- Optimize Marketing Spend: Focus on campaigns that yield high conversion rates to reduce CAC.

- Increase ARPU: Explore upselling and cross-selling opportunities to boost revenue per customer.

Reduce COGS: Streamline operational processes and negotiate better deals with software providers to lower expenses.

Improve Customer Retention: Enhance customer support and engagement to extend customer lifespan, thereby increasing LTV.

Monitor Unit Economics: Regularly evaluate these metrics to ensure the company's financial health and scalability.